### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by isha97**

### CODE REVIEW

File: spark-bigquery-parent/pom.xml [hunk 1]
```diff
@@ -52,21 +52,21 @@
...
-        <guava.version>33.0.0-jre</guava.version>
-        <jackson.version>2.16.2</jackson.version>
-        <netty.version>4.1.107.Final</netty.version>
+        <grpc.version>1.63.0</grpc.version>
+        <guava.version>33.1.0-jre</guava.version>
+        <jackson.version>2.17.0</jackson.version>
+        <netty.version>4.1.109.Final</netty.version>
         <paranamer.version>2.8</paranamer.version>
-        <protobuf.version>3.25.3</protobuf.version>
-        <zstd.version>1.4.9-1</zstd.version>
```
Comment (davidrabinowitz) [inline spark-bigquery-parent/pom.xml:69]: Please don't upgrade zstd - it is here as a provided dependency. 1.4.9 is the version Spark uses

**Review by davidrabinowitz — COMMENTED** [review]
Will approve once the IT pass

File: spark-bigquery-parent/pom.xml (hunk 1 above)
Comment (isha97) [reply spark-bigquery-parent/pom.xml:69]: Fixed

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**